services/agents/message_dispatcher.go

```diff
@@ -0,0 +1,38 @@
+package agents
+
+import (
+	"sync"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/percona/pmm/api/agent"
+)
+
+type messageDispatcher struct {
+	sync.RWMutex
+	l           *logrus.Entry
+	subscribers map[uint32]chan *agent.AgentMessage
+}
+
+func newMessageDispatcher(uuid string) *messageDispatcher {
+	return &messageDispatcher{
+		subscribers: make(map[uint32]chan *agent.AgentMessage),
+		l:           logrus.WithField("message dispatcher", uuid),
+	}
+}
+
+func (m *messageDispatcher) MessageReceived(message *agent.AgentMessage) {
+	m.Lock()
+	defer m.Unlock()
+	m.subscribers[message.Id] <- message
```

Comment:
We should gracefully handle unexpected messages, not panic with sending to `nil`.